VikingQuiz.Api/Controllers/QuizzesController.cs

```diff
@@ -109,82 +119,126 @@ public class QuizzesController : Controller
         [HttpPost("{id}")]
         [RequestSizeLimit(5_000_000)]
         [Authorize]
-        public async Task<IActionResult> Put(NewQuizViewModel quizBodyData, int id) {
+        public async Task<IActionResult> Put(NewQuizViewModel quizBodyData, int id)
+        {
+
+            IActionResult response = Ok();
             int currentUserId = User.Claims.GetUserId();
+
             IActionResult imageHttpResponse = FileValidityChecker(quizBodyData);
-            if (IActionResult.Equals(imageHttpResponse, Ok())) {
+            if (OkResult.Equals(imageHttpResponse, response) )
+            {
                 return imageHttpResponse;
             }
 
-            string fileUrl;
-            AzureBlobService blobService;
-            try {
-                blobService = new AzureBlobService();
-                await blobService.InitializeBlob();
-                var previousQuizState = quizRepository.GetQuizById(id);
-
-                blobService.DeletePhoto(previousQuizState.PictureUrl);
-                fileUrl = await blobService.UploadPhoto(quizBodyData.Files[0]);
-            }
-            catch (Exception) {
-                return BadRequest("Image could not be uploaded.");
+            UploadAndDeleteData uploadAndDeleteResult = await uploadAndDeletePhoto(id, quizBodyData.Files[0]);
+            if (OkResult.Equals(uploadAndDeleteResult.HttpResponseResult, Ok()) )
+            {
+                return uploadAndDeleteResult.HttpResponseResult;
             }
 
-            Quiz updatedQuiz = quizRepository.UpdateQuiz(new Quiz {
+   
+            IActionResult updateQuizResult = updateQuiz(new Quiz
+            {
                 Title = quizBodyData.Title,
-                PictureUrl = fileUrl,
+                PictureUrl = uploadAndDeleteResult.fileName,
                 UserId = currentUserId,
                 Id = id
             });
```

Comment:
Again, the thing with LastModified column which should be updated to current date and time